packages/restapi/src/lib/chat/helpers/user.ts

```diff
@@ -75,3 +75,62 @@ export const getConnectedUser = async (
     return { ...newUser, privateKey: decryptedPrivateKey };
   }
 };
+
+
+export const getConnectedUserV2 = async (
+  wallet: walletType,
+  privateKey: string | null,
+  env: string
+): Promise<IConnectedUser> => {
+  const address = await getAccountAddress(wallet);
+  const user = await get({ account: address, env: env || Constants.ENV.PROD });
+  if (user?.encryptedPrivateKey) {
+    if (privateKey) {
+      return { ...user, privateKey };
+    }
+    else {
+    let decryptedPrivateKey;
+    if (wallet.signer) {
+      decryptedPrivateKey = await decryptPGPKey({
+        signer: wallet.signer,
+        encryptedPGPPrivateKey: user.encryptedPrivateKey
+      })
+    } else {
+      decryptedPrivateKey = await decryptWithWalletRPCMethod(
+        user.encryptedPrivateKey,
+        address
+      );
+    }
```

Comment:
this part is also there from line 122-132, so maybe we can make a function to reuse it